### PR TITLE
STM32L4 low power timer fixes

### PR DESCRIPTION
**Pull request by agkaminski**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (stm32l4).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: proc/threads.c [hunk 1]
```diff
@@ -454,7 +453,8 @@ int threads_timeintr(unsigned int n, cpu_context_t *context, void *arg)
 
 	hal_spinlockClear(&threads_common.spinlock, &sc);
 
-	return EOK;
+	/* Invoke scheduler */
+	return 1;
```
Comment (xvuko) [inline proc/threads.c:457]: This invokes sheduler only on cpu 0 due to early return earlier.

Comment (agkaminski) [reply proc/threads.c:460]: Missed that, fixed, thanks!

File: hal/armv7m/stm32/l4/timer.c [hunk 2]
```diff
@@ -163,19 +165,45 @@ void timer_setAlarm(time_t us)
 		ticks = (ARR_VAL * 2) / 3;
```
Comment (xvuko) [inline hal/armv7m/stm32/l4/timer.c:160]: This is no longer needed as it implies that (setval > ARR_VAL)

Comment (agkaminski) [reply hal/armv7m/stm32/l4/timer.c]: Overflow check still needed, sightly reformatted the condition

**Review by astalke — APPROVED** [review]

**Review by anglov — APPROVED** [review]

**Pull request merged**